hoprd/hoprd/src/main.rs

```diff
@@ -287,7 +236,100 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
 
     processes.extend(hopr_processes.into_iter().map(|(k, v)| (HoprdProcesses::HoprLib(k), v)));
 
-    let mut signals = Signals::new([Signal::Hup, Signal::Int])?;
+    let signals = Signals::new([Signal::Hup, Signal::Int]).map_err(|e| HoprdError::OsError(e.to_string()))?;
+
+    Ok((signals, processes))
+}
+
+#[async_std::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let _ = init_logger();
+
+    let git_hash = option_env!("VERGEN_GIT_SHA").unwrap_or("unknown");
+    info!("This is HOPRd {} ({})", hopr_lib::constants::APP_VERSION, git_hash);
+
+    let args = <CliArgs as clap::Parser>::parse();
+    let should_dameonize = args.daemonize;
+    let cfg = hoprd::config::HoprdConfig::from_cli_args(args, false)?;
+
+    if std::env::var("DAPPNODE")
+        .map(|v| v.to_lowercase() == "true")
+        .unwrap_or(false)
+    {
+        info!("The HOPRd node appears to run on DappNode");
+    }
+
+    info!("Node configuration: {}", cfg.as_redacted_string()?);
+
+    if let hopr_lib::HostType::IPv4(address) = &cfg.hopr.host.address {
+        let ipv4 = std::net::Ipv4Addr::from_str(address).map_err(|e| HoprdError::ConfigError(e.to_string()))?;
+
+        if ipv4.is_loopback() && !cfg.hopr.transport.announce_local_addresses {
+            return Err(hopr_lib::errors::HoprLibError::GeneralError(
+                "Cannot announce a loopback address".into(),
+            ))?;
+        }
+    }
+
+    // Find or create an identity
+    let hopr_keys: HoprKeys = match &cfg.identity.private_key {
+        Some(private_key) => IdentityRetrievalModes::FromPrivateKey { private_key },
+        None => IdentityRetrievalModes::FromFile {
+            password: &cfg.identity.password,
+            id_path: &cfg.identity.file,
+        },
+    }
+    .try_into()?;
+
+    info!(
+        "This node has packet key '{}' and uses a blockchain address '{}'",
+        hopr_lib::Keypair::public(&hopr_keys.packet_key).to_peerid_str(),
+        hopr_lib::Keypair::public(&hopr_keys.chain_key).to_address().to_hex()
+    );
+
+    // TODO: the following check can be removed once [PR](https://github.com/hoprnet/hoprnet/pull/5665) is merged
+    if hopr_lib::Keypair::public(&hopr_keys.packet_key)
+        .to_string()
+        .starts_with("0xff")
+    {
+        warn!("This node uses an invalid packet key type and will not be able to become an effective relay node, please create a new identity!");
+    }
+
+    let (mut signals, processes) = if should_dameonize {
+        let daemonize = daemonize::Daemonize::new()
+            .working_directory(std::env::current_dir()?)
+            .user(
+                std::env::var("HOPRD_DAEMONIZE_USER")
+                    .unwrap_or("nobody".into())
+                    .as_str(),
+            )
+            .group(
+                std::env::var("HOPRD_DAEMONIZE_GROUP")
+                    .unwrap_or("nogroup".into())
```

Comment:
It should be `nobody` instead of `nogroup`.

Reply:
All of these are standard: `daemon`, `nobody` and `nogroup`.